fastfield_codecs/src/linearinterpol.rs

```diff
@@ -102,7 +102,11 @@ fn get_slope(first_val: u64, last_val: u64, num_vals: u64) -> f32 {
 
 #[inline]
 fn get_calculated_value(first_val: u64, pos: u64, slope: f32) -> u64 {
-    first_val + (pos as f32 * slope) as u64
+    if slope < 0.0 {
+        (first_val - (pos as f32 * slope) as u64) as u64
```

Comment:
i don't understand... why would that be correct?

Reply:
It's not, it should be
`(pos as f32 * -slope)`

Reply:
I moved this to the bugfix PR https://github.com/quickwit-oss/tantivy/pull/1439
This PR is now only adding a test and exposing the gcd

Reply:
Why is it * -slope?